### ISSUE TITLE
Add support for a system-wide shared libraries

### ISSUE THREAD

**Issue by blshkv** [avast-tl/retdec#356]
First of all, thank you for fixing the issue #94 , this is a good progress.
As the way of improvement, I would like to suggest implementing support of system-wide dependencies as well.  There are few tools which require capstone and all of them want to compile their own copy and link to it statically and there are a lot obvious disadvantages of such approach.

There is a way to verify if a system-wide library matches retdec requirements using pkg-config or cmake files and it should be possible to compile against them instead of compiling a static copy.
The tools which do not match can be still compiled locally (optionally).

The full list of the external tools is getting downloaded  to the following location: external/src

**Comment by PeterMatula** [comment]
Yes, this could be done. We might play work on this in the future, but it is not priority at the moment. If anyone would like to help in the meantime, please go on.

**Comment by blshkv** [comment]
I have tried to unbundle some packages but hit the very old llvm (3.0) compile quicky. It will take way to much changes to upgrade to v5 so I gave up at this point

**Comment by blshkv** [comment]
@PeterMatula
This is a good start:
https://github.com/avast/retdec/blob/master/deps/openssl/CMakeLists.txt
```RETDEC_FORCE_OPENSSL_BUILD```

Please implement similar variables for the remaining ```deps``` packages

**Comment by blshkv** [comment]
This is another short stopper:
https://github.com/avast/retdec/blob/master/src/bin2llvmir/utils/llvm.cpp#L382-L402
I'm unable to compile with system llvm just because of that part.
